src/contracts/middleware/StakeRegistryStorage.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../interfaces/IServiceManager.sol";
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "./VoteWeigherBase.sol";
+
+/**
+ * @title Storage variables for the `StakeRegistry` contract.
+ * @author Layr Labs, Inc.
+ * @notice This storage contract is separate from the logic to simplify the upgrade process.
+ */
+abstract contract StakeRegistryStorage is VoteWeigherBase, IStakeRegistry {
+    /// @notice the coordinator contract that this registry is associated with
+    IRegistryCoordinator public immutable registryCoordinator;
+
+    /// @notice In order to register for a quorum i, an operator must have at least `minimumStakeForQuorum[i]`
+    /// evaluated by this contract's 'VoteWeigher' logic.
+    uint96[256] public minimumStakeForQuorum;
+
+    /// @notice array of the history of the total stakes for each quorum -- marked as internal since getTotalStakeFromIndex is a getter for this
+    OperatorStakeUpdate[][256] internal _totalStakeHistory;
+
+    /// @notice mapping from operator's operatorId to the history of their stake updates
+    mapping(bytes32 => mapping(uint8 => OperatorStakeUpdate[])) public operatorIdToStakeHistory;
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator,
+        IStrategyManager _strategyManager,
+        IServiceManager _serviceManager
+    ) VoteWeigherBase(_strategyManager, _serviceManager)
+    // solhint-disable-next-line no-empty-blocks
+    {
+        registryCoordinator = _registryCoordinator;
+    }
+
+    // storage gap
+    uint256[50] private __GAP;
```

Comment:
not too sure on the proper size for this gap, but typically the idea is the total storage for a contract is a round number of slots.
looks like we're using 513 slots already -- maybe a gap size of 37 makes sense, so the total storage is a round 550? other obvious option is 87 slot gap to get 600 total slots

Reply:
looks like the `minimumStakeForQuorum` array actually gets packed into 128 slots, since it's using uint96s instead of uint256s. will make a commit to fix the gap size here.